lib/takeaway.rb

```diff
@@ -0,0 +1,40 @@
+class Takeaway 
+  attr_reader :basket, :menu, :grand_total
+    
+  def initialize 
+    @menu = {"chow mein": 1.99, "chicken balls": 2.00, "duck pancakes": 1.00}
+    @basket = []
+    @grand_total = 0.00
+  end
+    
+  def read_menu
+    @menu
+  end
+ 
+  def order(dish, quantity=1 )
+    @basket << {:dish => dish, :quantity => quantity, :total => total(dish, quantity)}
+    "You have added #{dish} x #{quantity}"
+  end
+ 
+  def total(dish, quantity)
```

Comment:
this method name could be a bit more informative - I wasn't immediately sure what it was doing on line 15